.devcontainer/devcontainer.json

```diff
@@ -1,17 +1,10 @@
 {
-  // TODO: Bump to ubuntu-24.04 once Python issues are fixed: https://github.com/argoproj/argo-workflows/pull/14302#issuecomment-2742883052
-  "image": "mcr.microsoft.com/vscode/devcontainers/base:ubuntu-22.04",
-  "appPort": 8080,
-  "features": {
-    "ghcr.io/devcontainers/features/go:1": {
-      "version": "1.23"
-    },
-    "ghcr.io/devcontainers/features/node:1": {
-      "version": "20"
-    },
-    "ghcr.io/devcontainers/features/docker-in-docker:2": {},
-    "ghcr.io/devcontainers/features/python:1": {}
-  },
+  // Dev container definition. Reference: https://containers.dev/implementors/json_reference/
+  "name": "Argo Workflows (pre-built)",
+
+  // This image is built and pushed by .github/workflows/devcontainer.yaml using .devcontainer/builder/devcontainer.json
+  "image": "quay.io/argoproj/argo-workflows-devcontainer",
```

Comment:
shall we version this? Although we don't change devcontianer that much but we want to make sure it works when we use a different argo version instead of figuring it how to make it work

Reply:
I thought about that, but unfortunately Dependabot doesn't support updating the `"image"` property in `devcontainer.json` yet (https://github.com/dependabot/dependabot-core/issues/7000), so there's no easy way of automating that yet AFAIK.

This image should work fine for older Argo releases, though I haven't tested that extensively. The only thing you might need to do manually is download an older `kubectl` version if you need to test old Kubernetes version. Note that the `pre-build.sh` script is not run as part of the image build process, so the cluster created by `k3d cluster create` is not part of the image.